### PR TITLE
Ensure working Fortran compiler if CMake needs to build OpenBLAS

### PR DESCRIPTION
**Pull request by stonecoldhughes**
Otherwise don't even care

edit: closes #211 .

### CODE REVIEW

File: contrib/contrib.cmake [hunk 1]
```diff
@@ -107,6 +107,17 @@ if (NOT LINALG_LIBS)
   else (NOT DEFINED LINALG_LIBS)
     message (STATUS "OpenBLAS not found. We'll build it from source.")
 
+    # OpenBLAS will build without Fortran, but it will silently neglect to build vital functions
+    include( CheckLanguage )
```
Comment (bmcdanie) [inline contrib/contrib.cmake]: does this just print the warning? I think we want to fail completely here, right?

Comment (stonecoldhughes) [reply contrib/contrib.cmake]: It does print the warning but I don't see any reason for it to fail there.

Comment (bmcdanie) [reply contrib/contrib.cmake]: the point of this PR is to make explicit the problems that occur when we build openblas without a fortran compiler. if we don't have fortran, but are trying to build openblas, we want to fail loudly - one warning message behind several screens of openblas compile barf won't help us.

Comment (grahamlopez) [reply contrib/contrib.cmake]: if it doesn't fail, won't the build just fail later on when trying to link? Generally, we'd like for things to fail at the cause, not the result.

Comment (bmcdanie) [reply contrib/contrib.cmake]: it doesn't help us to continue building under those conditions, right?

File: contrib/contrib.cmake [hunk 2]
```diff
@@ -107,6 +107,17 @@ if (NOT LINALG_LIBS)
...
+    message( STATUS "Checking for Fortran compiler..." )
+    check_language(Fortran)
+    if(CMAKE_Fortran_COMPILER)
+      enable_language(Fortran)
+      message( STATUS "Fortran compiler found" )
+    else()
+      message( FATAL_ERROR "Fortran compiler missing - required to build OpenBLAS" )
```
Comment (bmcdanie) [inline contrib/contrib.cmake]: or does this line cause build abort?

File: contrib/contrib.cmake (hunk 1 above)
Comment (bmcdanie) [reply contrib/contrib.cmake]: to be clear, when I say "fail here", I mean fail when we don't have a fortran compiler, not at this specific line.

File: contrib/contrib.cmake (hunk 2 above)
Comment (bmcdanie) [reply contrib/contrib.cmake]: @stonecoldhughes update - this is the line that causes abort.

**Pull request merged**